src/SW_Weather.c

```diff
@@ -511,18 +512,38 @@ void readAllWeather(
   Bool use_weathergenerator_only,
   char weather_prefix[]
 ) {
-    unsigned int yearIndex;
+    unsigned int yearIndex, year;
+
+    /* Interpolation is to be in base0 in `interpolate_monthlyValues()` */
+    Bool interpAsBase1 = swFALSE;
 
     for(yearIndex = 0; yearIndex < n_years; yearIndex++) {
+        year = yearIndex + startYear;
 
         // Set all daily weather values to missing
         _clear_hist_weather(allHist[yearIndex]);
 
+        // Update yearly day/month information needed when interpolating
+        // cloud cover, wind speed, and relative humidity if necessary
+        Time_new_year(year);
+
+        if(SW_Weather.use_cloudCoverMonthly) {

```

Comment:
`readAllWeather()` still uses globals "SW_Weather" and "SW_Sky" -- see earlier comment https://github.com/DrylandEcology/SOILWAT2/pull/338#discussion_r1066025836

Reply:
`_read_weather_hist()` still uses `SW_Weather`, etc.